remove_alma_inv_no_spaces.py

```diff
@@ -0,0 +1,196 @@
+        - last character "T", "M", or "R"
+        - no hyphen or the word "and"
+
+    If the call number meets these criteria, the space is removed.
+    If the call number does not meet these criteria, it is returned unchanged.
+
+    :param call_number: Original inventory number (852 $h)
+    :return: Updated inventory number with space removed, or the original if no update needed.
+    """
+
+    if (
+        len(call_number) >= 2
+        and call_number[-2] == " "
+        and call_number[-1] in {"T", "M", "R"}
+        and "-" not in call_number
+        and "and" not in call_number.lower()
+    ):
+        return call_number[:-2] + call_number[-1]
+    return call_number
+
+
+def main():
+    args = _get_arguments()
+    config = _get_config(args.config_file)
+    _configure_logging()
+
+    if args.environment == "production":
+        alma_api_key = config["alma_config"]["alma_api_key"]
+    else:  # sandbox
+        alma_api_key = config["alma_config"]["sandbox_alma_api_key"]
+
+    alma_client = AlmaAPIClient(api_key=alma_api_key)
+    logging.info(
+        f"Starting FTVA inventory number update process in {args.environment}."
+    )
+
+    alma_data = _read_alma_data(args.alma_data_file)
+
+    logging.info(f"Found {len(alma_data)} records to check for updates.")
+
+    # Itnitialize counter for updated records
+    updated_record_count = 0
+
+    for record in alma_data:
+        mms_id = record["MMS Id"]
+        holding_id = record["Holdings ID"]
+
+        try:
+            holding_record = alma_client.get_holding_record(mms_id, holding_id)
+            marc_record = holding_record.marc_record
+
+            if marc_record is None:
+                logging.warning(
+                    f"No MARC record found for MMS ID {mms_id}, holding {holding_id}; skipping"
+                )
+                continue
+
+            fields_852 = marc_record.get_fields("852")
+            if not fields_852:
+                logging.info(
+                    f"No 852 field found for MMS ID {mms_id}, holding {holding_id}; skipping"
+                )
+                continue
+
+            # Get the current 852 $h value (not repeatable, so take the first one)
+            current_inv_no = fields_852[0].get_subfields("h")[0]
+            new_inv_no = get_updated_call_number(current_inv_no)
+            if new_inv_no == current_inv_no:
+                logging.info(
+                    f"No update needed for MMS ID {mms_id}, holding {holding_id}"
+                    f" ('{current_inv_no}'); skipping"
+                )
+                continue
+
+            # Update the 852 $h value
+            fields_852[0].delete_subfield("h")
+            fields_852[0].add_subfield("h", new_inv_no)
+            logging.info(
+                f"Updated MMS ID {mms_id}, holding {holding_id}: "
+                f"'{current_inv_no}' -> '{new_inv_no}'"
+            )
+            updated_record_count += 1
```

Comment:
I recommend moving this increment to line 181, after `update_holding_record()`, so if the Alma update fails the counts are more accurate.